### PR TITLE
chore(grid-demo): various code cleanup, specifically around private `ClrDatagrid` members

### PR DESCRIPTION
**Pull request by kevinbuhmann**

### CODE REVIEW

File: grid-demo/src/app/angular/helpers/datagrid-private-member.helpers.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import { ElementRef } from '@angular/core';
+import { ClrDatagrid } from '@clr/angular';
+
+interface ClrDatagridPrivateMembers {
```
Comment (Jinnie) [inline grid-demo/src/app/angular/helpers/datagrid-private-member.helpers.ts:4]: This comment is general for the file. I see this adds some type safety, but we are still accessing private members.
What's the idea of this refactor? Do we gain anything long term?

File: grid-demo/src/app/angular/helpers/datagrid-active-cell.helpers.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+import { getDatagridKeyNavigationController } from './datagrid-private-member.helpers';
+
+const activeCellObservableCache = new WeakMap<ClrDatagrid, Observable<HTMLElement>>();
+
+export function fromActiveCell(datagrid: ClrDatagrid) {
+  return activeCellObservableCache.get(datagrid) || createAndCacheActiveCellObservable(datagrid);
+}
+
+function createAndCacheActiveCellObservable(datagrid: ClrDatagrid) {
+  const activeCellObservable = createActiveCellObservable(datagrid).pipe(publish(), refCount());
```
Comment (Jinnie) [inline grid-demo/src/app/angular/helpers/datagrid-active-cell.helpers.ts:14]: nit: publish+refCount seems to be deprecated. Looking at the docs, something like the following should be the suggested update:
```
createActiveCellObservable(datagrid).pipe(
  share({
    resetOnError: false,
    resetOnComplete: false,
    resetOnRefCountZero: false
  })
  ...
  ```

**Review by Jinnie — APPROVED** [review]
Thanks, looks nice and more readable now.
Got a few minor comments/questions below.

File: grid-demo/src/app/angular/helpers/datagrid-private-member.helpers.ts (hunk 1 above)
Comment (kevinbuhmann) [reply grid-demo/src/app/angular/helpers/datagrid-private-member.helpers.ts:4]: I just moved the `datagrid as any` casts to a single place and adds type safety. Hopefully, this all gets removed in the future, but I wanted to add some safety in the meantime.

File: grid-demo/src/app/angular/helpers/datagrid-active-cell.helpers.ts (hunk 2 above)
Comment (kevinbuhmann) [reply grid-demo/src/app/angular/helpers/datagrid-active-cell.helpers.ts:14]: Thanks!

Comment (kevinbuhmann) [reply grid-demo/src/app/angular/helpers/datagrid-active-cell.helpers.ts:14]: We aren't using RxJS 7.

Comment (kevinbuhmann) [reply grid-demo/src/app/angular/helpers/datagrid-active-cell.helpers.ts]: I went ahead and updated to RxJS 7.

Comment (Jinnie) [reply grid-demo/src/app/angular/helpers/datagrid-active-cell.helpers.ts]: You were right. Didn't notice we're behind. This was more of a question for my own learning. Thanks!

**Pull request merged**